modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java

```diff
@@ -0,0 +1,322 @@
+
+        String body = UserDefinedLocalAuthenticatorPayload.convertToJasonPayload(creationPayload);
+        body = StringUtils.replace(body, AuthenticatorPropertyConstants.AuthenticationType.IDENTIFICATION.toString(),
+                "InvalidAuthenticationType.");
+        Response response = getResponseOfPostNoFilter(AUTHENTICATOR_CUSTOM_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("code", equalTo("UE-10000"))
+                .body("message", equalTo("Invalid Request."))
+                .body("description", equalTo("Provided request body content is not in the expected format."));
+    }
+
+    @Test(priority = 7)
+    public void createUserDefinedLocalAuthenticatorInvalidAuthenticationType() throws JsonProcessingException {
+
+        String body = UserDefinedLocalAuthenticatorPayload.convertToJasonPayload(creationPayload);
+        body = StringUtils.replace(body, AuthenticationType.TypeEnum.BASIC.toString(), "InvalidType.");
+        Response response = getResponseOfPostNoFilter(AUTHENTICATOR_CUSTOM_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("code", equalTo("UE-10000"))
+                .body("message", equalTo("Invalid Request."))
+                .body("description", equalTo("Provided request body content is not in the expected format."));
+    }
+
+    @Test(priority = 8)
+    public void updateNonExistingUserDefinedLocalAuthenticator() throws JsonProcessingException {
+
+        String body = UserDefinedLocalAuthenticatorPayload.convertToJasonPayload(updatePayload);
+        Response response = getResponseOfPutWithNoFilter(AUTHENTICATOR_CUSTOM_API_BASE_PATH + PATH_SEPARATOR +
+                customIdPId, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_NOT_FOUND);
+    }
+
+
+    @Test(priority = 9)
+    public void createUserDefinedLocalAuthenticatorWithExistingAuthenticatorName() throws JsonProcessingException {
+
+        String body = UserDefinedLocalAuthenticatorPayload.convertToJasonPayload(creationPayload);
+        Response response = getResponseOfPost(AUTHENTICATOR_CUSTOM_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        Response responseForDuplication = getResponseOfPost(AUTHENTICATOR_CUSTOM_API_BASE_PATH, body);
+        responseForDuplication.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("code", equalTo("AUT-60003"))
```

Comment:
The same error code is duplicated

Reply:
Will be fix with https://github.com/wso2/product-is/issues/22050